components/builder-db/src/models/origin.rs

```diff
@@ -121,12 +129,16 @@ impl Origin {
     }
 
     pub fn delete(origin: &str, conn: &PgConnection) -> QueryResult<()> {
+        // By this point, most of the associated origin data has already been manually deleted
+        // by the user. We ensure this by double checking the most critical tables are already empty
+        // via builder_api::server::resources::origins::origin_delete_preflight
+        // The transaction entries below explicitly enumerate deletion of all table data associated
+        // with the origin to ensure no vestigial data remains.
+        //
+        // TODO: Add origin deletion request into an origin audit table.
```

Comment:
👍  We should file an issue so we can pull it into our work queue and link to it here

Reply:
I've created the issue and placed the URL into the comments in https://github.com/habitat-sh/builder/pull/1298/commits/25ace14b200f0dcaa374bb405fa01df009ca5fde